launcher/src/main/java/org/apache/spark/launcher/ChildProcAppHandle.java

```diff
@@ -103,67 +46,27 @@ public synchronized void kill() {
       try {
         childProc.exitValue();
       } catch (IllegalThreadStateException e) {
-        // Child is still alive. Try to use Java 8's "destroyForcibly()" if available,
-        // fall back to the old API if it's not there.
         try {
-          Method destroy = childProc.getClass().getMethod("destroyForcibly");
-          destroy.invoke(childProc);
-        } catch (Exception inner) {
           childProc.destroy();
+        } catch (Exception inner) {
```

Comment:
This will never happen. You're breaking the existing functionality. See my previous comment (https://github.com/apache/spark/pull/15009/files#r84788125).